# 6. 분석

## 과제1. 어린이 보호구역 외 어린이 교통사고 위험지역 20개소 제시
### 6-1) 격자별 교통사고 위험지수 생성
> - **목표:** R 프로그래밍을 사용하여 최적 회귀 모형을 선정한 뒤 회귀계수를 이용하여 **격자별 교통사고 위험지수(hazard_index)**를 설정한다. 
> - **y: 어린이 교통 사고 횟수, X: 교통 사고 요인** 
> - **변수 선택:** step AIC로 주요 변수를 선택한 포아송 모형과 음이항 모형의 AIC를 비교하여 모형을 선택하고 그 모형에서 유의수준 0.05 이하의 유의한 변수를 선택한다.
> - **모형 선택:** **y가 사고 횟수로 영과잉 자료**이므로 영과잉 포아송 모형과 영과잉 음이항 모형을 **AIC로 비교하여 최적 모형을 선택**한다.
> - **위험 지수 설정:** **최적 모형의 회귀계수를 가중치로 사용**하여 격자별 교통사고 위험지수를 설정한다. 

In [13]:
library(readxl)
library(MASS)
library(pscl)
library(xlsx)

In [2]:
all_data = read_excel('input/all_data.xlsx')
all_data<-as.data.frame(all_data)
data<-all_data[,-c(1:6)]
data<- subset(data, select=-c(building_area_28))
head(data,3)

New names:
* `` -> ...1



accident_cnt_02 parking_01 car_cnt_03 pop_total_04 pop_child_05
1 0               0          0          0            0           
2 0               0          0          0            0           
3 0               0          0          0            0           
  ratio_child_0405 pop_working_06 pop_elder_07 schoolzone_cctv_cnt_09
1 0                0              0            0                     
2 0                0              0            0                     
3 0                0              0            0                     
  road_width_schoolzone_09 ⋯ link_length_24 conjestion_freq_25
1 0                        ⋯ 0              0                 
2 0                        ⋯ 0              0                 
3 0                        ⋯ 0              0                 
  conjestion_time_26 bad_bld_27 good_bld_27 intersection_23 link_cnt_23
1 0                  0          0           0               0          
2 0                  0          0           0               0          
3 0                  0          0           0               0          
  tunnel_bridge_23 oneway_23 barrier_23
1 0                0         0         
2 0                0         0         
3 0                0         0

### 6-1-1) 포아송 모형

In [20]:
poisson<-glm(accident_cnt_02~., data=data, family=poisson(link="log"))
summary(poisson)

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Call:
glm(formula = accident_cnt_02 ~ ., family = poisson(link = "log"), 
    data = data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.5321  -0.3124  -0.2344  -0.2321   4.5756  

Coefficients: (3 not defined because of singularities)
                           Estimate Std. Error z value Pr(>|z|)    
(Intercept)              -3.614e+00  9.328e-02 -38.751  < 2e-16 ***
parking_01                1.796e-03  6.938e-04   2.589 0.009624 ** 
car_cnt_03               -1.210e-04  2.838e-04  -0.426 0.669864    
pop_total_04              5.458e-02  2.524e-02   2.162 0.030615 *  
pop_child_05             -6.472e-02  2.528e-02  -2.560 0.010454 *  
ratio_child_0405          2.011e+00  9.750e-01   2.063 0.039126 *  
pop_working_06           -5.366e-02  2.534e-02  -2.117 0.034220 *  
pop_elder_07             -3.471e-02  2.520e-02  -1.378 0.168350    
schoolzone_cctv_cnt_09   -1.562e-01  2.887e-01  -0.541 0.588524    
road_width_schoolzone_09 -9.856e-03  5.462e-02  -0.180 0.8

In [21]:
stepAIC(poisson) 

Start:  AIC=2469.1
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    road_width_schoolzone_09 + kindergarten_cnt_13 + manlesscctv_cnt_15 + 
    sign_cnt_16 + crosswalk_cnt_17 + bump_cnt_18 + trafficlight_cnt_19 + 
    cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + 
    road_rank_23 + road_length_23 + road_width_23 + road_numcross_23 + 
    building_cnt_27 + building_level_up_27 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    pop_fluid_1416_08 + schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + 
    total_traffic_810_24 + total_traffic_1416_24 + link_length_24 + 
    conjestion_freq_25 + conjestion_time_26 + bad_bld_27 + good_bld_27 + 
    intersection_23 + link_cnt_23 + tunnel_bridge_23 + oneway_23 + 
    barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 


Step:  AIC=2469.1
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    road_width_schoolzone_09 + kindergarten_cnt_13 + manlesscctv_cnt_15 + 
    sign_cnt_16 + crosswalk_cnt_17 + bump_cnt_18 + trafficlight_cnt_19 + 
    cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + 
    road_rank_23 + road_length_23 + road_width_23 + road_numcross_23 + 
    building_cnt_27 + building_level_up_27 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    pop_fluid_1416_08 + schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + 
    total_traffic_810_24 + total_traffic_1416_24 + link_length_24 + 
    conjestion_freq_25 + conjestion_time_26 + bad_bld_27 + good_bld_27 + 
    intersection_23 + tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 


Step:  AIC=2469.1
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    road_width_schoolzone_09 + kindergarten_cnt_13 + manlesscctv_cnt_15 + 
    sign_cnt_16 + crosswalk_cnt_17 + bump_cnt_18 + trafficlight_cnt_19 + 
    cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + 
    road_rank_23 + road_length_23 + road_width_23 + road_numcross_23 + 
    building_cnt_27 + building_level_up_27 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    pop_fluid_1416_08 + schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + 
    total_traffic_810_24 + link_length_24 + conjestion_freq_25 + 
    conjestion_time_26 + bad_bld_27 + good_bld_27 + intersection_23 + 
    tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 


Step:  AIC=2469.1
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    road_width_schoolzone_09 + kindergarten_cnt_13 + manlesscctv_cnt_15 + 
    sign_cnt_16 + crosswalk_cnt_17 + bump_cnt_18 + trafficlight_cnt_19 + 
    cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + 
    road_rank_23 + road_length_23 + road_width_23 + road_numcross_23 + 
    building_cnt_27 + building_level_up_27 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + total_traffic_810_24 + 
    link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    bad_bld_27 + good_bld_27 + intersection_23 + tunnel_bridge_23 + 
    oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                           Df Deviance    AIC
- manlesscctv_cnt_15        1   1599.5 2467.1
- bump_cnt_18               1   1599.5 2467.1
- building_level_up_27      1   1599.5 2467.1
- road_width_schoolzone_09  1   1599.5 2467.1
- building_cnt_27           1   1599.5 2467.1
- total_traffic_810_24      1   1599.6 2467.2
- kindergarten_cnt_13       1   1599.6 2467.2
- car_cnt_03                1   1599.7 2467.3
- schoolzone_9              1   1599.7 2467.3
- schoolzone_cctv_cnt_09    1   1599.8 2467.4
- good_bld_27               1   1599.8 2467.4
- road_rank_23              1   1599.9 2467.5
- bad_bld_27                1   1600.0 2467.6
- tunnel_bridge_23          1   1600.4 2468.1
- link_length_24            1   1600.5 2468.1
- building_level_down_27    1   1600.5 2468.1
- oneway_23                 1   1601.1 2468.8
- academy_cnt_30            1   1601.1 2468.8
- road_numcross_23          1   1601.2 2468.8
- conjestion_time_26        1   1601.3 2468.9
- pop_elder_07              1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2467.1
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    road_width_schoolzone_09 + kindergarten_cnt_13 + sign_cnt_16 + 
    crosswalk_cnt_17 + bump_cnt_18 + trafficlight_cnt_19 + cctv_cnt_20 + 
    bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + road_rank_23 + 
    road_length_23 + road_width_23 + road_numcross_23 + building_cnt_27 + 
    building_level_up_27 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + schoolzone_9 + 
    school_cnt_10 + sidewalk_cnt_21 + total_traffic_810_24 + 
    link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    bad_bld_27 + good_bld_27 + intersection_23 + tunnel_bridge_23 + 
    oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                           Df Deviance    AIC
- bump_cnt_18               1   1599.5 2465.1
- building_level_up_27      1   1599.5 2465.1
- road_width_schoolzone_09  1   1599.5 2465.1
- building_cnt_27           1   1599.5 2465.1
- total_traffic_810_24      1   1599.6 2465.2
- kindergarten_cnt_13       1   1599.6 2465.2
- car_cnt_03                1   1599.7 2465.3
- schoolzone_9              1   1599.7 2465.3
- schoolzone_cctv_cnt_09    1   1599.8 2465.4
- good_bld_27               1   1599.8 2465.4
- road_rank_23              1   1599.9 2465.5
- bad_bld_27                1   1600.0 2465.6
- tunnel_bridge_23          1   1600.4 2466.1
- link_length_24            1   1600.5 2466.1
- building_level_down_27    1   1600.5 2466.1
- oneway_23                 1   1601.1 2466.8
- road_numcross_23          1   1601.2 2466.8
- academy_cnt_30            1   1601.2 2466.8
- conjestion_time_26        1   1601.3 2466.9
- pop_elder_07              1   1601.3 2466.9
- road_length_23            1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2465.11
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    road_width_schoolzone_09 + kindergarten_cnt_13 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_rank_23 + road_length_23 + 
    road_width_23 + road_numcross_23 + building_cnt_27 + building_level_up_27 + 
    building_level_down_27 + sports_cnt_29 + academy_cnt_30 + 
    bicycleroad_21 + pop_fluid_810_08 + schoolzone_9 + school_cnt_10 + 
    sidewalk_cnt_21 + total_traffic_810_24 + link_length_24 + 
    conjestion_freq_25 + conjestion_time_26 + bad_bld_27 + good_bld_27 + 
    intersection_23 + tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                           Df Deviance    AIC
- building_level_up_27      1   1599.5 2463.1
- road_width_schoolzone_09  1   1599.5 2463.1
- building_cnt_27           1   1599.5 2463.1
- total_traffic_810_24      1   1599.6 2463.2
- kindergarten_cnt_13       1   1599.6 2463.2
- car_cnt_03                1   1599.7 2463.3
- schoolzone_9              1   1599.7 2463.3
- schoolzone_cctv_cnt_09    1   1599.8 2463.4
- good_bld_27               1   1599.8 2463.5
- road_rank_23              1   1599.9 2463.5
- bad_bld_27                1   1600.0 2463.6
- tunnel_bridge_23          1   1600.4 2464.1
- link_length_24            1   1600.5 2464.1
- building_level_down_27    1   1600.5 2464.2
- oneway_23                 1   1601.2 2464.8
- road_numcross_23          1   1601.2 2464.8
- academy_cnt_30            1   1601.2 2464.8
- conjestion_time_26        1   1601.3 2464.9
- pop_elder_07              1   1601.3 2465.0
- road_length_23            1   1601.5 2465.1
<none>                          15

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2463.12
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    road_width_schoolzone_09 + kindergarten_cnt_13 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_rank_23 + road_length_23 + 
    road_width_23 + road_numcross_23 + building_cnt_27 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + total_traffic_810_24 + 
    link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    bad_bld_27 + good_bld_27 + intersection_23 + tunnel_bridge_23 + 
    oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                           Df Deviance    AIC
- road_width_schoolzone_09  1   1599.5 2461.2
- total_traffic_810_24      1   1599.6 2461.2
- building_cnt_27           1   1599.6 2461.2
- kindergarten_cnt_13       1   1599.6 2461.2
- car_cnt_03                1   1599.7 2461.3
- schoolzone_9              1   1599.7 2461.3
- schoolzone_cctv_cnt_09    1   1599.8 2461.4
- road_rank_23              1   1599.9 2461.5
- bad_bld_27                1   1600.0 2461.6
- good_bld_27               1   1600.0 2461.6
- tunnel_bridge_23          1   1600.5 2462.1
- link_length_24            1   1600.5 2462.1
- oneway_23                 1   1601.2 2462.8
- road_numcross_23          1   1601.2 2462.8
- academy_cnt_30            1   1601.2 2462.8
- conjestion_time_26        1   1601.3 2462.9
- pop_elder_07              1   1601.3 2463.0
- road_length_23            1   1601.5 2463.1
<none>                          1599.5 2463.1
- road_width_23             1   1601.8 2463.4
- road_cnt_23               1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2461.15
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    kindergarten_cnt_13 + sign_cnt_16 + crosswalk_cnt_17 + trafficlight_cnt_19 + 
    cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + 
    road_rank_23 + road_length_23 + road_width_23 + road_numcross_23 + 
    building_cnt_27 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + schoolzone_9 + 
    school_cnt_10 + sidewalk_cnt_21 + total_traffic_810_24 + 
    link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    bad_bld_27 + good_bld_27 + intersection_23 + tunnel_bridge_23 + 
    oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- total_traffic_810_24    1   1599.6 2459.2
- kindergarten_cnt_13     1   1599.6 2459.3
- building_cnt_27         1   1599.6 2459.3
- car_cnt_03              1   1599.7 2459.3
- schoolzone_9            1   1599.8 2459.4
- schoolzone_cctv_cnt_09  1   1599.8 2459.4
- road_rank_23            1   1599.9 2459.5
- bad_bld_27              1   1600.0 2459.6
- good_bld_27             1   1600.0 2459.7
- tunnel_bridge_23        1   1600.5 2460.1
- link_length_24          1   1600.5 2460.1
- road_numcross_23        1   1601.2 2460.8
- academy_cnt_30          1   1601.2 2460.8
- oneway_23               1   1601.2 2460.8
- pop_elder_07            1   1601.4 2461.0
- conjestion_time_26      1   1601.4 2461.0
- road_length_23          1   1601.5 2461.1
<none>                        1599.5 2461.2
- road_width_23           1   1601.8 2461.5
- road_cnt_23             1   1602.6 2462.2
- conjestion_freq_25      1   1602.6 2462.3
- barrier_23              1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2459.25
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    kindergarten_cnt_13 + sign_cnt_16 + crosswalk_cnt_17 + trafficlight_cnt_19 + 
    cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + 
    road_rank_23 + road_length_23 + road_width_23 + road_numcross_23 + 
    building_cnt_27 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + schoolzone_9 + 
    school_cnt_10 + sidewalk_cnt_21 + link_length_24 + conjestion_freq_25 + 
    conjestion_time_26 + bad_bld_27 + good_bld_27 + intersection_23 + 
    tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- kindergarten_cnt_13     1   1599.7 2457.3
- building_cnt_27         1   1599.7 2457.3
- car_cnt_03              1   1599.8 2457.4
- schoolzone_9            1   1599.8 2457.4
- schoolzone_cctv_cnt_09  1   1599.9 2457.5
- road_rank_23            1   1600.0 2457.6
- bad_bld_27              1   1600.1 2457.7
- good_bld_27             1   1600.1 2457.8
- tunnel_bridge_23        1   1600.7 2458.3
- link_length_24          1   1600.8 2458.4
- oneway_23               1   1601.3 2458.9
- road_numcross_23        1   1601.3 2458.9
- pop_elder_07            1   1601.4 2459.0
- conjestion_time_26      1   1601.5 2459.1
- academy_cnt_30          1   1601.5 2459.1
- road_length_23          1   1601.6 2459.2
<none>                        1599.6 2459.2
- road_width_23           1   1601.9 2459.5
- road_cnt_23             1   1602.6 2460.2
- conjestion_freq_25      1   1602.8 2460.4
- barrier_23              1   1603.4 2461.0
- ratio_child_0405        1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2457.35
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    sign_cnt_16 + crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + 
    bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + road_rank_23 + 
    road_length_23 + road_width_23 + road_numcross_23 + building_cnt_27 + 
    building_level_down_27 + sports_cnt_29 + academy_cnt_30 + 
    bicycleroad_21 + pop_fluid_810_08 + schoolzone_9 + school_cnt_10 + 
    sidewalk_cnt_21 + link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    bad_bld_27 + good_bld_27 + intersection_23 + tunnel_bridge_23 + 
    oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- building_cnt_27         1   1599.8 2455.4
- car_cnt_03              1   1599.9 2455.6
- schoolzone_cctv_cnt_09  1   1600.0 2455.7
- schoolzone_9            1   1600.1 2455.7
- road_rank_23            1   1600.1 2455.7
- bad_bld_27              1   1600.2 2455.8
- good_bld_27             1   1600.2 2455.8
- tunnel_bridge_23        1   1600.8 2456.4
- link_length_24          1   1600.9 2456.6
- oneway_23               1   1601.5 2457.1
- pop_elder_07            1   1601.5 2457.1
- road_numcross_23        1   1601.5 2457.1
- conjestion_time_26      1   1601.6 2457.2
- road_length_23          1   1601.7 2457.3
<none>                        1599.7 2457.3
- academy_cnt_30          1   1601.8 2457.4
- road_width_23           1   1602.1 2457.7
- road_cnt_23             1   1602.6 2458.2
- conjestion_freq_25      1   1602.9 2458.6
- barrier_23              1   1603.5 2459.2
- pop_working_06          1   1603.9 2459.5
- bicycleroad_21          1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2455.45
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    sign_cnt_16 + crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + 
    bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + road_rank_23 + 
    road_length_23 + road_width_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + link_length_24 + 
    conjestion_freq_25 + conjestion_time_26 + bad_bld_27 + good_bld_27 + 
    intersection_23 + tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- car_cnt_03              1   1600.0 2453.7
- schoolzone_cctv_cnt_09  1   1600.2 2453.8
- schoolzone_9            1   1600.2 2453.8
- road_rank_23            1   1600.2 2453.8
- bad_bld_27              1   1600.3 2453.9
- good_bld_27             1   1600.6 2454.2
- tunnel_bridge_23        1   1600.9 2454.5
- link_length_24          1   1601.0 2454.7
- road_numcross_23        1   1601.6 2455.2
- oneway_23               1   1601.6 2455.2
- pop_elder_07            1   1601.6 2455.2
- conjestion_time_26      1   1601.7 2455.3
- road_length_23          1   1601.8 2455.4
<none>                        1599.8 2455.4
- academy_cnt_30          1   1601.9 2455.5
- road_width_23           1   1602.1 2455.8
- road_cnt_23             1   1602.8 2456.4
- conjestion_freq_25      1   1603.0 2456.6
- barrier_23              1   1603.7 2457.3
- pop_working_06          1   1604.0 2457.7
- bicycleroad_21          1   1604.2 2457.8
- pop_total_04            1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2453.66
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    sign_cnt_16 + crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + 
    bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + road_rank_23 + 
    road_length_23 + road_width_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + link_length_24 + 
    conjestion_freq_25 + conjestion_time_26 + bad_bld_27 + good_bld_27 + 
    intersection_23 + tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- schoolzone_cctv_cnt_09  1   1600.4 2452.0
- schoolzone_9            1   1600.4 2452.0
- road_rank_23            1   1600.4 2452.1
- bad_bld_27              1   1600.5 2452.1
- good_bld_27             1   1600.8 2452.4
- tunnel_bridge_23        1   1601.1 2452.7
- link_length_24          1   1601.3 2452.9
- oneway_23               1   1601.8 2453.4
- pop_elder_07            1   1601.8 2453.4
- road_numcross_23        1   1601.8 2453.5
- conjestion_time_26      1   1601.9 2453.5
- academy_cnt_30          1   1602.0 2453.6
- road_length_23          1   1602.0 2453.6
<none>                        1600.0 2453.7
- road_width_23           1   1602.3 2453.9
- road_cnt_23             1   1603.0 2454.6
- conjestion_freq_25      1   1603.2 2454.8
- barrier_23              1   1603.9 2455.5
- pop_working_06          1   1604.3 2455.9
- bicycleroad_21          1   1604.4 2456.0
- ratio_child_0405        1   1604.4 2456.0
- bus_cnt_22              1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2451.99
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_rank_23 + road_length_23 + 
    road_width_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + link_length_24 + 
    conjestion_freq_25 + conjestion_time_26 + bad_bld_27 + good_bld_27 + 
    intersection_23 + tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- schoolzone_9            1   1600.4 2450.1
- road_rank_23            1   1600.8 2450.4
- bad_bld_27              1   1600.8 2450.4
- good_bld_27             1   1601.1 2450.8
- tunnel_bridge_23        1   1601.4 2451.0
- link_length_24          1   1601.7 2451.3
- oneway_23               1   1602.1 2451.7
- road_numcross_23        1   1602.2 2451.8
- conjestion_time_26      1   1602.2 2451.8
- pop_elder_07            1   1602.2 2451.9
- academy_cnt_30          1   1602.3 2451.9
- road_length_23          1   1602.3 2452.0
<none>                        1600.4 2452.0
- road_width_23           1   1602.6 2452.2
- road_cnt_23             1   1603.3 2452.9
- conjestion_freq_25      1   1603.4 2453.0
- barrier_23              1   1604.4 2454.0
- ratio_child_0405        1   1604.7 2454.4
- pop_working_06          1   1604.8 2454.4
- intersection_23         1   1604.8 2454.4
- bus_cnt_22              1   1604.8 2454.5
- bicycleroad_21          1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2450.05
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_rank_23 + road_length_23 + 
    road_width_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    school_cnt_10 + sidewalk_cnt_21 + link_length_24 + conjestion_freq_25 + 
    conjestion_time_26 + bad_bld_27 + good_bld_27 + intersection_23 + 
    tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- road_rank_23            1   1600.8 2448.4
- bad_bld_27              1   1600.9 2448.5
- good_bld_27             1   1601.2 2448.8
- tunnel_bridge_23        1   1601.4 2449.1
- link_length_24          1   1601.7 2449.3
- oneway_23               1   1602.2 2449.8
- conjestion_time_26      1   1602.2 2449.8
- road_numcross_23        1   1602.3 2449.9
- academy_cnt_30          1   1602.3 2450.0
- pop_elder_07            1   1602.3 2450.0
- road_length_23          1   1602.4 2450.0
<none>                        1600.4 2450.1
- road_width_23           1   1602.6 2450.2
- road_cnt_23             1   1603.4 2451.0
- conjestion_freq_25      1   1603.4 2451.0
- barrier_23              1   1604.6 2452.2
- ratio_child_0405        1   1604.8 2452.4
- intersection_23         1   1604.8 2452.5
- bus_cnt_22              1   1604.9 2452.5
- pop_working_06          1   1604.9 2452.5
- bicycleroad_21          1   1604.9 2452.5
- pop_total_04            1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2448.44
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_width_23 + 
    road_numcross_23 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    bad_bld_27 + good_bld_27 + intersection_23 + tunnel_bridge_23 + 
    oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- bad_bld_27              1   1601.3 2446.9
- good_bld_27             1   1601.6 2447.2
- tunnel_bridge_23        1   1601.8 2447.4
- link_length_24          1   1602.0 2447.6
- oneway_23               1   1602.2 2447.8
- road_length_23          1   1602.5 2448.1
- conjestion_time_26      1   1602.6 2448.2
- academy_cnt_30          1   1602.7 2448.3
- pop_elder_07            1   1602.7 2448.3
- road_numcross_23        1   1602.7 2448.3
<none>                        1600.8 2448.4
- road_width_23           1   1603.1 2448.7
- conjestion_freq_25      1   1603.8 2449.4
- road_cnt_23             1   1603.9 2449.5
- barrier_23              1   1604.8 2450.4
- intersection_23         1   1605.2 2450.8
- ratio_child_0405        1   1605.2 2450.8
- pop_working_06          1   1605.2 2450.8
- bicycleroad_21          1   1605.3 2450.9
- pop_total_04            1   1605.4 2451.0
- bus_cnt_22              1   1605.4 2451.0
- parking_01              1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2446.9
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_width_23 + 
    road_numcross_23 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    good_bld_27 + intersection_23 + tunnel_bridge_23 + oneway_23 + 
    barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- good_bld_27             1   1602.0 2445.7
- tunnel_bridge_23        1   1602.2 2445.8
- link_length_24          1   1602.5 2446.1
- oneway_23               1   1602.7 2446.3
- road_length_23          1   1602.9 2446.6
- conjestion_time_26      1   1603.1 2446.7
- pop_elder_07            1   1603.1 2446.8
- academy_cnt_30          1   1603.1 2446.8
- road_numcross_23        1   1603.2 2446.8
<none>                        1601.3 2446.9
- road_width_23           1   1603.5 2447.2
- conjestion_freq_25      1   1604.3 2447.9
- road_cnt_23             1   1604.3 2447.9
- barrier_23              1   1605.3 2448.9
- ratio_child_0405        1   1605.5 2449.1
- intersection_23         1   1605.6 2449.2
- pop_working_06          1   1605.6 2449.3
- bicycleroad_21          1   1605.8 2449.4
- pop_total_04            1   1605.8 2449.4
- bus_cnt_22              1   1605.8 2449.5
- parking_01              1   1606.1 2449.7
- sidewalk_cnt_21         1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2445.66
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_width_23 + 
    road_numcross_23 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    intersection_23 + tunnel_bridge_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- tunnel_bridge_23        1   1603.0 2444.6
- link_length_24          1   1603.2 2444.8
- oneway_23               1   1603.4 2445.0
- road_length_23          1   1603.7 2445.3
- pop_elder_07            1   1603.9 2445.5
- road_numcross_23        1   1603.9 2445.5
- conjestion_time_26      1   1603.9 2445.5
- academy_cnt_30          1   1603.9 2445.6
<none>                        1602.0 2445.7
- road_width_23           1   1604.4 2446.0
- road_cnt_23             1   1605.1 2446.7
- conjestion_freq_25      1   1605.1 2446.7
- barrier_23              1   1606.0 2447.6
- ratio_child_0405        1   1606.3 2447.9
- pop_working_06          1   1606.4 2448.0
- bicycleroad_21          1   1606.4 2448.0
- intersection_23         1   1606.4 2448.0
- pop_total_04            1   1606.6 2448.2
- bus_cnt_22              1   1606.7 2448.3
- parking_01              1   1606.8 2448.4
- sidewalk_cnt_21         1   1607.3 2448.9
- pop_child_05            1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2444.61
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_width_23 + 
    road_numcross_23 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + link_length_24 + conjestion_freq_25 + conjestion_time_26 + 
    intersection_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- link_length_24          1   1604.0 2443.7
- oneway_23               1   1604.3 2443.9
- road_length_23          1   1604.7 2444.3
- pop_elder_07            1   1604.7 2444.3
- conjestion_time_26      1   1604.8 2444.5
- road_numcross_23        1   1604.9 2444.5
- academy_cnt_30          1   1604.9 2444.6
<none>                        1603.0 2444.6
- road_width_23           1   1605.2 2444.9
- conjestion_freq_25      1   1605.9 2445.5
- road_cnt_23             1   1606.7 2446.3
- barrier_23              1   1607.0 2446.6
- ratio_child_0405        1   1607.0 2446.7
- intersection_23         1   1607.2 2446.8
- pop_working_06          1   1607.2 2446.8
- bicycleroad_21          1   1607.2 2446.8
- pop_total_04            1   1607.3 2446.9
- bus_cnt_22              1   1607.4 2447.1
- parking_01              1   1607.8 2447.4
- sidewalk_cnt_21         1   1608.0 2447.6
- pop_child_05            1   1609.0 2448.6
- road_maxspeed_23        1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2443.67
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_width_23 + 
    road_numcross_23 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + conjestion_time_26 + 
    intersection_23 + oneway_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- oneway_23               1   1605.4 2443.0
- road_length_23          1   1605.7 2443.3
- pop_elder_07            1   1605.8 2443.4
- conjestion_time_26      1   1605.8 2443.4
- road_numcross_23        1   1605.9 2443.6
- academy_cnt_30          1   1605.9 2443.6
<none>                        1604.0 2443.7
- road_width_23           1   1606.3 2443.9
- conjestion_freq_25      1   1606.8 2444.4
- road_cnt_23             1   1607.9 2445.5
- bicycleroad_21          1   1608.0 2445.6
- ratio_child_0405        1   1608.1 2445.7
- intersection_23         1   1608.2 2445.8
- pop_working_06          1   1608.2 2445.8
- barrier_23              1   1608.3 2445.9
- pop_total_04            1   1608.3 2446.0
- bus_cnt_22              1   1608.5 2446.1
- parking_01              1   1608.8 2446.5
- sidewalk_cnt_21         1   1609.4 2447.0
- pop_child_05            1   1610.0 2447.6
- road_maxspeed_23        1   1611.5 2449.1
- school_cnt_10           1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2442.98
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_width_23 + 
    road_numcross_23 + building_level_down_27 + sports_cnt_29 + 
    academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + conjestion_time_26 + 
    intersection_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- academy_cnt_30          1   1606.7 2442.3
- conjestion_time_26      1   1607.1 2442.7
- pop_elder_07            1   1607.2 2442.8
- road_numcross_23        1   1607.3 2442.9
<none>                        1605.4 2443.0
- road_length_23          1   1607.6 2443.2
- conjestion_freq_25      1   1608.0 2443.6
- road_width_23           1   1608.2 2443.8
- road_cnt_23             1   1609.2 2444.8
- ratio_child_0405        1   1609.4 2445.0
- pop_working_06          1   1609.7 2445.3
- bicycleroad_21          1   1609.7 2445.4
- intersection_23         1   1609.8 2445.4
- pop_total_04            1   1609.9 2445.5
- parking_01              1   1610.3 2445.9
- bus_cnt_22              1   1610.3 2445.9
- barrier_23              1   1610.3 2446.0
- sidewalk_cnt_21         1   1610.6 2446.2
- pop_child_05            1   1611.6 2447.2
- road_maxspeed_23        1   1611.6 2447.2
- school_cnt_10           1   1616.4 2452.1
- building_level_down_27  1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2442.34
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_width_23 + 
    road_numcross_23 + building_level_down_27 + sports_cnt_29 + 
    bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + sidewalk_cnt_21 + 
    conjestion_freq_25 + conjestion_time_26 + intersection_23 + 
    barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- conjestion_time_26      1   1608.2 2441.8
- pop_elder_07            1   1608.3 2441.9
- road_numcross_23        1   1608.6 2442.2
<none>                        1606.7 2442.3
- road_length_23          1   1608.9 2442.5
- conjestion_freq_25      1   1609.2 2442.8
- road_width_23           1   1609.3 2442.9
- road_cnt_23             1   1610.3 2443.9
- pop_working_06          1   1610.7 2444.3
- bicycleroad_21          1   1610.7 2444.3
- pop_total_04            1   1610.8 2444.4
- intersection_23         1   1611.1 2444.7
- ratio_child_0405        1   1611.1 2444.7
- bus_cnt_22              1   1611.2 2444.8
- parking_01              1   1611.9 2445.5
- sidewalk_cnt_21         1   1612.0 2445.6
- barrier_23              1   1612.4 2446.0
- pop_child_05            1   1612.6 2446.2
- road_maxspeed_23        1   1612.6 2446.2
- school_cnt_10           1   1617.9 2451.5
- building_level_down_27  1   1619.0 2452.6
- crosswalk_cnt_17        1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2441.85
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_width_23 + 
    road_numcross_23 + building_level_down_27 + sports_cnt_29 + 
    bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + sidewalk_cnt_21 + 
    conjestion_freq_25 + intersection_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- road_width_23           1   1609.4 2441.0
- pop_elder_07            1   1609.6 2441.2
- conjestion_freq_25      1   1609.8 2441.4
<none>                        1608.2 2441.8
- road_length_23          1   1610.2 2441.9
- road_numcross_23        1   1610.6 2442.2
- road_cnt_23             1   1611.3 2442.9
- pop_working_06          1   1611.8 2443.4
- pop_total_04            1   1612.0 2443.6
- ratio_child_0405        1   1612.2 2443.8
- bus_cnt_22              1   1612.4 2444.0
- bicycleroad_21          1   1612.5 2444.1
- intersection_23         1   1613.0 2444.6
- parking_01              1   1613.0 2444.6
- sidewalk_cnt_21         1   1613.6 2445.2
- pop_child_05            1   1613.6 2445.2
- road_maxspeed_23        1   1614.0 2445.6
- barrier_23              1   1615.8 2447.4
- school_cnt_10           1   1619.2 2450.8
- building_level_down_27  1   1620.0 2451.7
- crosswalk_cnt_17        1   1622.6 2454.2
- trafficlight_cnt_19     1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2441.04
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_cnt_23 + road_maxspeed_23 + road_length_23 + road_numcross_23 + 
    building_level_down_27 + sports_cnt_29 + bicycleroad_21 + 
    pop_fluid_810_08 + school_cnt_10 + sidewalk_cnt_21 + conjestion_freq_25 + 
    intersection_23 + barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
- pop_elder_07            1   1611.2 2440.8
- road_length_23          1   1611.4 2441.0
<none>                        1609.4 2441.0
- road_numcross_23        1   1611.6 2441.3
- conjestion_freq_25      1   1611.8 2441.4
- bus_cnt_22              1   1613.0 2442.6
- intersection_23         1   1613.1 2442.7
- bicycleroad_21          1   1613.3 2443.0
- pop_working_06          1   1613.6 2443.2
- pop_total_04            1   1613.8 2443.4
- ratio_child_0405        1   1613.8 2443.5
- road_cnt_23             1   1613.9 2443.6
- parking_01              1   1614.6 2444.2
- pop_child_05            1   1615.5 2445.1
- sidewalk_cnt_21         1   1615.6 2445.2
- school_cnt_10           1   1620.9 2450.5
- barrier_23              1   1621.5 2451.1
- road_maxspeed_23        1   1622.0 2451.6
- building_level_down_27  1   1622.1 2451.8
- crosswalk_cnt_17        1   1624.2 2453.8
- trafficlight_cnt_19     1   1630.8 2460.4
- sports_cnt_29           1   16

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Step:  AIC=2440.81
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + sign_cnt_16 + crosswalk_cnt_17 + 
    trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + 
    road_maxspeed_23 + road_length_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + intersection_23 + 
    barrier_23



Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message:
“glm.fit: fitted rates 

                         Df Deviance    AIC
<none>                        1611.2 2440.8
- road_length_23          1   1613.2 2440.8
- road_numcross_23        1   1613.2 2440.8
- conjestion_freq_25      1   1613.4 2441.0
- intersection_23         1   1614.3 2441.9
- ratio_child_0405        1   1614.9 2442.5
- bus_cnt_22              1   1615.2 2442.8
- bicycleroad_21          1   1615.2 2442.8
- road_cnt_23             1   1616.1 2443.8
- parking_01              1   1616.4 2444.0
- sidewalk_cnt_21         1   1617.5 2445.1
- school_cnt_10           1   1622.5 2450.1
- barrier_23              1   1622.9 2450.5
- building_level_down_27  1   1623.6 2451.2
- road_maxspeed_23        1   1624.0 2451.6
- crosswalk_cnt_17        1   1626.2 2453.8
- pop_working_06          1   1628.1 2455.7
- trafficlight_cnt_19     1   1631.9 2459.6
- sports_cnt_29           1   1633.2 2460.8
- pop_total_04            1   1635.4 2463.1
- pop_fluid_810_08        1   1637.8 2465.4
- sign_cnt_16             1   16


Call:  glm(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + sign_cnt_16 + crosswalk_cnt_17 + 
    trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + 
    road_maxspeed_23 + road_length_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + intersection_23 + 
    barrier_23, family = poisson(link = "log"), data = data)

Coefficients:
           (Intercept)              parking_01            pop_total_04  
             -3.578651                0.001835                0.019506  
          pop_child_05        ratio_child_0405          pop_working_06  
             -0.029456                1.826065               -0.018609  
           sign_cnt_16        crosswalk_cnt_17     trafficlight_cnt_19  
              0.239978                0.140828                0.109475  
           cctv_cnt_20              bus_

- step AIC를 통해 변수 선택 한 최종 포아송 모형 

In [23]:
pois<-  glm(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + sign_cnt_16 + crosswalk_cnt_17 + 
    trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + 
    road_maxspeed_23 + road_length_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + intersection_23 + 
    barrier_23, family = poisson(link = "log"), data = data)

summary(pois)

Warning message:
“glm.fit: fitted rates numerically 0 occurred”



Call:
glm(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + sign_cnt_16 + crosswalk_cnt_17 + 
    trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + 
    road_maxspeed_23 + road_length_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + intersection_23 + 
    barrier_23, family = poisson(link = "log"), data = data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.3767  -0.3175  -0.2387  -0.2363   4.5010  

Coefficients:
                         Estimate Std. Error z value Pr(>|z|)    
(Intercept)            -3.579e+00  8.798e-02 -40.675  < 2e-16 ***
parking_01              1.835e-03  6.896e-04   2.661 0.007789 ** 
pop_total_04            1.951e-02  3.734e-03   5.223 1.76e-07 ***
pop_child_05           -2.946e-02  4.983e-03  -5.911 3.39e-09 ***
ratio_child_0405        1.826e+00  

In [24]:
AIC(pois)

[1] 2440.812

###  6-1-2) 음이항 모형

In [26]:
nb<-glm.nb(accident_cnt_02~., data=data) 
summary(nb)

Warning message:
“glm.fit: fitted rates numerically 0 occurred”
Warning message in glm.nb(accident_cnt_02 ~ ., data = data):
“alternation limit reached”



Call:
glm.nb(formula = accident_cnt_02 ~ ., data = data, init.theta = 1.080290318, 
    link = log)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.9937  -0.2955  -0.2162  -0.2137   3.3901  

Coefficients: (3 not defined because of singularities)
                           Estimate Std. Error z value Pr(>|z|)    
(Intercept)              -3.769e+00  1.037e-01 -36.344  < 2e-16 ***
parking_01                1.941e-03  8.825e-04   2.200 0.027808 *  
car_cnt_03               -1.015e-05  3.012e-04  -0.034 0.973121    
pop_total_04              5.360e-02  3.053e-02   1.755 0.079210 .  
pop_child_05             -6.348e-02  3.052e-02  -2.080 0.037512 *  
ratio_child_0405          2.456e+00  1.170e+00   2.099 0.035826 *  
pop_working_06           -5.271e-02  3.068e-02  -1.718 0.085858 .  
pop_elder_07             -3.260e-02  3.042e-02  -1.072 0.283936    
schoolzone_cctv_cnt_09   -8.885e-02  3.600e-01  -0.247 0.805032    
road_width_schoolzone_09 -2.163e-02  6.887e-02  -

In [27]:
stepAIC(nb) 

Start:  AIC=2418.34
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + pop_elder_07 + schoolzone_cctv_cnt_09 + 
    road_width_schoolzone_09 + kindergarten_cnt_13 + manlesscctv_cnt_15 + 
    sign_cnt_16 + crosswalk_cnt_17 + bump_cnt_18 + trafficlight_cnt_19 + 
    cctv_cnt_20 + bus_cnt_22 + road_cnt_23 + road_maxspeed_23 + 
    road_rank_23 + road_length_23 + road_width_23 + road_numcross_23 + 
    building_cnt_27 + building_level_up_27 + building_level_down_27 + 
    sports_cnt_29 + academy_cnt_30 + bicycleroad_21 + pop_fluid_810_08 + 
    pop_fluid_1416_08 + schoolzone_9 + school_cnt_10 + sidewalk_cnt_21 + 
    total_traffic_810_24 + total_traffic_1416_24 + link_length_24 + 
    conjestion_freq_25 + conjestion_time_26 + bad_bld_27 + good_bld_27 + 
    intersection_23 + link_cnt_23 + tunnel_bridge_23 + oneway_23 + 
    barrier_23


Step:  AIC=2418.34
accident_cnt_02 ~ parking_01 + car_cnt_03 + pop_total_04 + pop_child_05 

Warning message in glm.nb(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + :
“alternation limit reached”


                         Df    AIC
- tunnel_bridge_23        1 2386.9
- link_length_24          1 2386.9
- academy_cnt_30          1 2387.1
- road_length_23          1 2387.4
<none>                      2387.6
- road_numcross_23        1 2387.9
- parking_01              1 2388.3
- sidewalk_cnt_21         1 2388.3
- bicycleroad_21          1 2388.4
- ratio_child_0405        1 2388.8
- conjestion_freq_25      1 2390.1
- barrier_23              1 2391.0
- building_level_down_27  1 2391.1
- bus_cnt_22              1 2393.5
- sports_cnt_29           1 2394.0
- school_cnt_10           1 2394.7
- road_maxspeed_23        1 2394.7
- trafficlight_cnt_19     1 2395.3
- pop_working_06          1 2397.1
- crosswalk_cnt_17        1 2398.4
- sign_cnt_16             1 2400.8
- pop_fluid_810_08        1 2401.5
- pop_total_04            1 2402.7
- pop_child_05            1 2407.7
- intersection_23         1 2417.4
- cctv_cnt_20             1 2431.1

Step:  AIC=2386.92
accident_cnt_02 ~ parking_01 + pop_

Warning message in glm.nb(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + :
“alternation limit reached”


                         Df    AIC
- road_length_23          1 2385.3
<none>                      2385.5
- road_numcross_23        1 2385.7
- bicycleroad_21          1 2385.9
- sidewalk_cnt_21         1 2386.2
- ratio_child_0405        1 2386.8
- parking_01              1 2386.9
- conjestion_freq_25      1 2388.0
- building_level_down_27  1 2388.8
- barrier_23              1 2389.6
- bus_cnt_22              1 2391.0
- trafficlight_cnt_19     1 2392.5
- school_cnt_10           1 2392.7
- road_maxspeed_23        1 2393.0
- pop_working_06          1 2395.0
- crosswalk_cnt_17        1 2398.2
- pop_fluid_810_08        1 2399.8
- sports_cnt_29           1 2399.9
- sign_cnt_16             1 2400.1
- pop_total_04            1 2400.6
- pop_child_05            1 2405.6
- intersection_23         1 2415.4
- cctv_cnt_20             1 2428.7

Step:  AIC=2385.32
accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + sign_cnt_16 + crosswalk_cnt_17 + 
    

Warning message in glm.nb(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + :
“alternation limit reached”


                         Df    AIC
<none>                      2385.3
- road_numcross_23        1 2385.4
- bicycleroad_21          1 2385.6
- sidewalk_cnt_21         1 2386.1
- ratio_child_0405        1 2386.6
- parking_01              1 2386.7
- conjestion_freq_25      1 2387.9
- building_level_down_27  1 2388.6
- barrier_23              1 2389.8
- bus_cnt_22              1 2390.9
- road_maxspeed_23        1 2392.0
- trafficlight_cnt_19     1 2392.3
- school_cnt_10           1 2392.5
- pop_working_06          1 2394.7
- crosswalk_cnt_17        1 2398.0
- sports_cnt_29           1 2399.6
- sign_cnt_16             1 2399.8
- pop_fluid_810_08        1 2399.9
- pop_total_04            1 2400.2
- pop_child_05            1 2405.4
- intersection_23         1 2415.8
- cctv_cnt_20             1 2428.7



Call:  glm.nb(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + 
    pop_child_05 + ratio_child_0405 + pop_working_06 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_maxspeed_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + intersection_23 + 
    barrier_23, data = data, init.theta = 1.031629867, link = log)

Coefficients:
           (Intercept)              parking_01            pop_total_04  
             -3.745824                0.001911                0.020694  
          pop_child_05        ratio_child_0405          pop_working_06  
             -0.031019                2.290981               -0.019689  
           sign_cnt_16        crosswalk_cnt_17     trafficlight_cnt_19  
              0.286906                0.190271                0.101839  
           cctv_cnt_20              bus_cnt_22        road_max

- step AIC를 통해 변수 선택 한 최종 음이항 모형 

In [9]:
nbinom<-glm.nb(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + 
    pop_child_05 + ratio_child_0405 + pop_working_06 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_maxspeed_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + intersection_23 + 
    barrier_23, data = data, init.theta = 1.031629867, link = log)
summary(nbinom)


Call:
glm.nb(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + 
    pop_child_05 + ratio_child_0405 + pop_working_06 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_maxspeed_23 + road_numcross_23 + building_level_down_27 + 
    sports_cnt_29 + bicycleroad_21 + pop_fluid_810_08 + school_cnt_10 + 
    sidewalk_cnt_21 + conjestion_freq_25 + intersection_23 + 
    barrier_23, data = data, init.theta = 1.031681092, link = log)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.8801  -0.2999  -0.2189  -0.2161   3.3653  

Coefficients:
                         Estimate Std. Error z value Pr(>|z|)    
(Intercept)            -3.7458139  0.0990781 -37.807  < 2e-16 ***
parking_01              0.0019112  0.0008745   2.185 0.028864 *  
pop_total_04            0.0206941  0.0045692   4.529 5.93e-06 ***
pop_child_05           -0.0310188  0.0060070  -5.164 2.42e-07 ***
ratio_child_0405        2.2909970  1.1363059   2.016 0.04

In [29]:
AIC(nbinom) 

[1] 2387.317

## 분석 결과

#### step AIC로 주요 변수를 선택한 포아송 모형의 AIC는 2441, 음이항 모형의 AIC는 2387이므로 **AIC 기준으로 음이항 모형을 선택**하고 **유의수준  0.05 이하의 유의한 변수를 선택**한다.

#### 어린이 교통 사고 횟수를 설명하는데 유의한 변수로 **선택된 변수는 총 18개**이다. 
- parking_01: 격자내 주정차 단속 횟수 
- pop_total_04: 격자내 총인구 
- pop_child_05: 격자내 유소년 인구 
- ratio_child_0405: 격자내 유소년인구/총인구
- pop_working_06: 격자내 생산가능인구
- sign_cnt_16: 격자내 도로안전표지 개수 
- crosswalk_cnt_17: 격자내 횡단보도 개수
- trafficlight_cnt_19: 격자내 신호등 개수 
- cctv_cnt_20: 격자내 cctv개수
- bus_cnt_22:격자내 버스정류장 개수 
- road_maxspeed_23: 격자내 도로의 제한속도 
- building_level_down_27: 격자내 건물의 지하층수
- sports_cnt_29: 격자내 체육시설 개수 
- pop_fluid_810_08: 2019년 오전 8~10시 격자내 유동인구
- school_cnt_10: 격자내 초등학교 개수 
- conjestion_freq_25: 격자내 혼잡빈도강도  
- intersection_23: 격자내 교차로 개수 
- barrier_23: 격자내 중앙분리대 개수 

## 영과잉(Zero-inflated) 모델
- ZIP(Zero Inflated Poisson Regression), ZINB(Zero Inflated Negative Binomial Regression)
- 기존의 포아송회귀모형이나 음이항회귀모형과 같은 전통적인 추론방법은 특정한 값에서 기대도수보다 관측도수가 많이 나타나는 자료에 대하여 잘 설명할 수 없음
- 교통사고 자료와 같이 영의 비율이 기존의 포아송회귀모형 또는 음이항회귀모형에 의해 기대되는 영의 비율보다 높게 관측되는 자료를 영과잉 자료라고함.
- 영과잉 모형은 자료에서 과도하게 나타나는 영을 설명하기 위해 영에 대한 점확률을 가지는 분포를 기존의 분포와 합성한 것임

### 6-1-3) 영과잉 포아송 모형 (ZIP 모형)

In [4]:
zip<-zeroinfl( accident_cnt_02 ~ parking_01 + pop_total_04 + 
    pop_child_05 + ratio_child_0405 + pop_working_06 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_maxspeed_23  + building_level_down_27 + 
    sports_cnt_29  + pop_fluid_810_08 + school_cnt_10 + conjestion_freq_25 + intersection_23 + 
    barrier_23 |1, data=data, dist="poisson")
summary(zip)

Warning message in sqrt(diag(object$vcov)):
“NaNs produced”



Call:
zeroinfl(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + sign_cnt_16 + crosswalk_cnt_17 + 
    trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + road_maxspeed_23 + 
    building_level_down_27 + sports_cnt_29 + pop_fluid_810_08 + school_cnt_10 + 
    conjestion_freq_25 + intersection_23 + barrier_23 | 1, data = data, 
    dist = "poisson")

Pearson residuals:
    Min      1Q  Median      3Q     Max 
-1.2630 -0.2292 -0.1707 -0.1692 11.4832 

Count model coefficients (poisson with log link):
                         Estimate Std. Error z value Pr(>|z|)    
(Intercept)            -3.2254648  0.1165002 -27.686  < 2e-16 ***
parking_01              0.0020064  0.0007777   2.580 0.009888 ** 
pop_total_04            0.0165242         NA      NA       NA    
pop_child_05           -0.0314970         NA      NA       NA    
ratio_child_0405        2.5057935  0.8781514   2.853 0.004324 ** 
pop_working_06         -0.0141729       

In [5]:
AIC(zip)

[1] 2437.039

### 6-1-4) 영과잉 음이항 모형 (ZINB 모형)

In [6]:
zinb<-zeroinfl( accident_cnt_02 ~ parking_01 + pop_total_04 + 
    pop_child_05 + ratio_child_0405 + pop_working_06 + sign_cnt_16 + 
    crosswalk_cnt_17 + trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + 
    road_maxspeed_23  + building_level_down_27 + 
    sports_cnt_29  + pop_fluid_810_08 + school_cnt_10 + conjestion_freq_25 + intersection_23 + 
    barrier_23 |1, data=data, dist="negbin")
summary(zinb)


Call:
zeroinfl(formula = accident_cnt_02 ~ parking_01 + pop_total_04 + pop_child_05 + 
    ratio_child_0405 + pop_working_06 + sign_cnt_16 + crosswalk_cnt_17 + 
    trafficlight_cnt_19 + cctv_cnt_20 + bus_cnt_22 + road_maxspeed_23 + 
    building_level_down_27 + sports_cnt_29 + pop_fluid_810_08 + school_cnt_10 + 
    conjestion_freq_25 + intersection_23 + barrier_23 | 1, data = data, 
    dist = "negbin")

Pearson residuals:
    Min      1Q  Median      3Q     Max 
-0.9087 -0.2111 -0.1543 -0.1525 12.5900 

Count model coefficients (negbin with log link):
                        Estimate Std. Error z value Pr(>|z|)    
(Intercept)            -3.737255   0.104279 -35.839  < 2e-16 ***
parking_01              0.002047   0.001063   1.925 0.054271 .  
pop_total_04            0.021192   0.005148   4.116 3.85e-05 ***
pop_child_05           -0.032021   0.006456  -4.959 7.07e-07 ***
ratio_child_0405        2.540878   1.241960   2.046 0.040770 *  
pop_working_06         -0.020205   0.006036  -3.

In [7]:
AIC(zinb)

[1] 2393.626

### 분석 결과 
- step AIC로 주요 변수를 선택한 영과잉 포아송 모형의 AIC는 2437, 영과잉 음이항 모형의 AIC는 2394이므로 **AIC 기준으로 영과잉 음이항 모형을 최적 모형으로 선택**한다. 

### 6-1-5) 격자별 교통사고 위험지수 생성
> - 영과잉 음이항 모형을 통해 얻은 회귀 계수를 가중치로 사용하여 격자별 위험 지수 설정한다. 

In [12]:
data_index<-all_data  

In [14]:
# exponential 변환
data_index$hazard_index=exp(summary(zinb)$coefficients$count[1])+
                 exp(summary(zinb)$coefficients$count[3])*all_data$pop_total_04+
                 exp(summary(zinb)$coefficients$count[4])*all_data$pop_child_05+
                 exp(summary(zinb)$coefficients$count[5])*all_data$ratio_child_0405+
                 exp(summary(zinb)$coefficients$count[6])*all_data$pop_working_06 +
                 exp(summary(zinb)$coefficients$count[7])*all_data$sign_cnt_16+
                 exp(summary(zinb)$coefficients$count[8])*all_data$crosswalk_cnt_17+
                 exp(summary(zinb)$coefficients$count[9])*all_data$trafficlight_cnt_19+
                 exp(summary(zinb)$coefficients$count[10])*all_data$cctv_cnt_20+
                 exp(summary(zinb)$coefficients$count[11])*all_data$bus_cnt_22 +
                 exp(summary(zinb)$coefficients$count[12])*all_data$road_maxspeed_23+
                 exp(summary(zinb)$coefficients$count[13])*all_data$building_level_down_27+
                 exp(summary(zinb)$coefficients$count[14])*all_data$sports_cnt_29 +
                 exp(summary(zinb)$coefficients$count[15])*all_data$pop_fluid_810_08 +
                 exp(summary(zinb)$coefficients$count[16])*all_data$school_cnt_10+
                 exp(summary(zinb)$coefficients$count[17])*all_data$conjestion_freq_25+
                 exp(summary(zinb)$coefficients$count[18])*all_data$intersection_23+
                 exp(summary(zinb)$coefficients$count[19])*all_data$barrier_23 

In [18]:
write.xlsx(data_index,'input/data_index.xlsx')  # 위험지수를 포함한 데이터를 저장